src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java

```diff
@@ -3244,6 +3277,71 @@ protected SingleChronicleQueueBuilder builder(@NotNull File file, @NotNull WireT
+        SetTimeProvider timeProvider = new SetTimeProvider();
+        timeProvider.currentTimeMillis(System.currentTimeMillis() - 2_000);
+        final File dir = getTmpDir();
+        final RollCycles rollCycle = RollCycles.TEST_SECONDLY;
+
+        // write first message
+        try (ChronicleQueue queue =
+                     binary(dir)
+                             .rollCycle(rollCycle).timeProvider(timeProvider).build()) {
+            ExcerptAppender excerptAppender = queue.acquireAppender();
+            excerptAppender.writeText("someText");
+
+            ExecutorService executorService = Executors.newFixedThreadPool(2,
+                    new NamedThreadFactory("test"));
+
+            Future<?> f1 = executorService.submit(() -> {
+
+                try (ChronicleQueue queue2 = binary(dir)
+                        .rollCycle(rollCycle).timeProvider(timeProvider).build()) {
+                    queue2.acquireAppender().writeText("someText more");
+                }
+                timeProvider.advanceMillis(1100);
+                try (ChronicleQueue queue2 = binary(dir)
+                        .rollCycle(rollCycle).timeProvider(timeProvider).build()) {
+                    queue2.acquireAppender().writeText("someText more");
+                }
+            });
+
+            Future<?> f2 = executorService.submit(() -> {
+
+                // write second message
+                try (ChronicleQueue queue2 = binary(dir)
+                        .rollCycle(rollCycle).timeProvider(timeProvider).build()) {
+
+                    for (int i = 0; i < 5; i++) {
+                        queue2.acquireAppender().writeText("someText more");
+                        timeProvider.advanceMillis(400);
+                    }
+                }
+            });
+
+            f1.get(10, TimeUnit.SECONDS);
+            // System.out.println(queue.dump().replaceAll("(?m)^#.+$\\n", ""));
+            f2.get(10, TimeUnit.SECONDS);
+
+            executorService.shutdownNow();
+        }
+    }
+
+    @NotNull
+    protected SingleChronicleQueueBuilder builderWithAppendListener(@NotNull File file, @NotNull WireType wireType) {
+        appenderListenerDump.clear();
+        return SingleChronicleQueueBuilder.builder(file, wireType)
+                .rollCycle(RollCycles.TEST4_DAILY)
+                .timeProvider(new SetTimeProvider("2021/11/17T12:34:56").advanceMillis(1000))
+                .appenderListener((w, idx) -> {
+                    appenderListenerDump.append("idx: ").append(Long.toHexString(idx)).append("\n");
+                    w.bytes().readSkip(-4);
+                    appenderListenerDump.append(Wires.fromSizePrefixedBlobs(w)).append("\n");
+                })
+                .testBlockSize();
+    }
+
     @NotNull
     protected SingleChronicleQueueBuilder binary(@NotNull File file) {
         return builder(file, WireType.BINARY_LIGHT);
```

Comment:
This diff is impossible to review... is there a way we could do it with less change?